File: Reference-Android-Java/app/src/main/java/xyz/fay/reference/networking/response/CityResponse.java
package xyz.fay.reference.networking.response;

import xyz.fay.parcel.Parcelable;
import xyz.fay.parcel.Parcelize;

@Parcelize
public final class CityResponse extends Parcelable {
    private final CityListResponse[] cities;

    public CityResponse(CityListResponse[] cities) {
        this.cities = cities;
    }
}

File: Reference-Android-Kotlin/app/src/main/java/xyz/fay/reference/networking/response/CityResponse.kt
package xyz.fay.reference.networking.response

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class CityResponse(
        val cities: List<CityListResponse>
) : Parcelable

@Parcelize
data class CityListResponse(
        val name: String
) : Parcelable
